extensions/amp-analytics/0.1/events.js

```diff
@@ -661,15 +676,41 @@ class TimerEventHandler {
   /**
    * @param {!Window} win
    */
-  clearInterval(win) {
+  stopTimer_(win) {
     if (!this.isRunning()) {
       return;
     }
     win.clearInterval(this.intervalId_);
     this.intervalId_ = undefined;
+    this.calculateDuration_();
+    this.lastRequestTime_ = undefined;
+    this.intervalCallback_();
+    this.intervalCallback_ = null;
     this.unlistenForStop_();
     this.listenForStart_();
   }
+
+  /** @private */
+  calculateDuration_() {
+    if (this.startTime_) {
+      this.timerDuration_ +=
+          Date.now() - (this.lastRequestTime_ || this.startTime_);
+    }
+  }
+
+  /** @return {{timerDuration: number, timerStart: number}} */
+  getTimerVars() {
+    if (this.isRunning()) {
+      this.calculateDuration_();
+      this.lastRequestTime_ = Date.now();
+    }
+    const reportDuration = this.timerDuration_;
+    this.timerDuration_ = 0;
```

Comment:
do you really need `timerDuration` to be a member field?

Reply:
Reorganized to remove need for it.